### PR TITLE
IMP: Python-based command blocks

### PR DESCRIPTION
**Pull request by thermokarst**
@ebolyen, can you take a look at this when you get a chance? I'm not quite sure why I am not able to catch stdout from my python blocks, maybe I just missed something simple. Thanks!

### CODE REVIEW

File: source/sphinx_extensions/command_block/extension.py [hunk 1]
```diff
@@ -128,13 +139,26 @@ def run(self):
...
+            'shell': self._execute_sh_commands,
+            'python': self._execute_py_commands,
+        }[runtime](commands, working_dir)
+
+    def _execute_py_commands(self, commands, working_dir):
+        with tempfile.NamedTemporaryFile(mode='w', encoding='utf-8') as fh:
+            for command in commands:
+                fh.write('%s\n' % command)
+            return self._execute_sh_commands(
+                ['%s %s' % (sys.executable, str(fh.name))], working_dir)
```
Comment (ebolyen) [inline source/sphinx_extensions/command_block/extension.py]: Is there ever a need to share scope across blocks for this?

Comment (thermokarst) [reply source/sphinx_extensions/command_block/extension.py]: Probably, and that is certainly cumbersome w this current approach (would require Artifacts to be saved to disk).

**Comment by ebolyen** [comment]
When you say stdout isn't working, is it like `print` isn't working? I don't see how it would fail...

**Comment by ebolyen** [comment]
Alternatively, would a doctest directive work instead? We used that in the dev docs and it generally works pretty well.

**Comment by ebolyen** [comment]
In particular: https://www.sphinx-doc.org/en/master/usage/extensions/doctest.html#directive-testcode might do what you need.

**Comment by thermokarst** [comment]
Yeah, none of my print statements will work. Will look into the `testcode` directive, that might get me there.

**Comment by thermokarst** [comment]
Okay, that directive doesn't quite do it for me. I wanted this functionality for the Artifact API section of my WIP "Utilities" tutorial, but I think instead I will just skip that interface (for now). Maybe @Oddant1 wants to take over this PR?

**Pull request closed without merging**